app/src/main/java/com/example/pokedex/ui/PokedexListViewModel.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.pokedex.ui
+
+import android.os.Parcelable
+import androidx.lifecycle.*
+import com.example.pokedex.DI
+import com.example.pokedex.domain.ListItem
+import com.example.pokedex.domain.PokemonList
+import com.example.pokedex.usecase.GetAllPokemonUseCase
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import kotlinx.parcelize.Parcelize
+
+private const val STATE_KEY = "POKEDEX_LIST_STATE_KEY"
+
+class PokedexListViewModel @JvmOverloads constructor(
+    private val savedStateHandle: SavedStateHandle,
+    private val getAllPokemonUseCase: GetAllPokemonUseCase = DI.instance.getAllPokemonUseCase
+): ViewModel() {
+    init {
+        loadPokemonList()
+    }
+
+    val viewStateFlow: Flow<PokedexListViewState> by lazy {
+        stateFlow.map { it.toViewState() }.distinctUntilChanged()
+    }
+
+    private val stateFlow: MutableStateFlow<PokedexListState> = MutableStateFlow(
+        savedStateHandle.get(STATE_KEY) ?: PokedexListState(
+            pokemonList = null
+        )
+    )
+
+    private fun loadPokemonList() {
+        viewModelScope.launch {
+            stateFlow.value = PokedexListState(getAllPokemonUseCase.execute())
```

Comment:
(nit) In this case it doesn't matter a ton, because your state only has one item. However, it is typical to use the `copy` function rather than recreating a new PokedexListState from the constructor
```
stateFlow.value = stateFlow.value.copy(pokemonList = getAllPokemonUseCase.execute())
```

Reply:
Done